lms/djangoapps/courseware/date_summary.py

```diff
@@ -78,51 +75,25 @@ def __init__(self, course, user):
         self.course = course
         self.user = user
 
-    def get_context(self):
-        """Return the template context used to render this summary block."""
-        return {
-            'title': self.title,
-            'date': self._format_date(),
-            'description': self.description,
-            'css_class': self.css_class,
-            'link': self.link,
-            'link_text': self.link_text,
-        }
-
-    def render(self):
+    @property
+    def relative_datestring(self):
         """
-        Return an HTML representation of this summary block.
+        Generates a fuzzy description of the time until (or past) self.date.
+        Translators: 'relative_datestring' is a fuzzy description of the time until
+        'course_date.date'.
+        For example, 'course_date.date' might be "2021-01-01 10:26:31+00:00",
+        and if today were December 5th, 2020, 'relative' would be "1 month".
         """
-        return render_to_string('courseware/date_summary.html', self.get_context())
 
-    def _format_date(self):
-        """
-        Return this block's date in a human-readable format. If the date
-        is None, returns the empty string.
-        """
         if self.date is None:
             return ''
-        locale = to_locale(get_language())
         delta = self.date - datetime.now(utc)
-        try:
-            relative_date = format_timedelta(delta, locale=locale)
-        # Babel doesn't have translations for Esperanto, so we get
-        # a KeyError when testing translations with
-        # ?preview-lang=eo. This should not happen with any other
-        # languages. See https://github.com/python-babel/babel/issues/107
-        except KeyError:
-            relative_date = format_timedelta(delta)
+        relative_date = format_timedelta(delta)
         date_has_passed = delta.days < 0
-        # Translators: 'absolute' is a date such as "Jan 01,
-        # 2020". 'relative' is a fuzzy description of the time until
-        # 'absolute'. For example, 'absolute' might be "Jan 01, 2020",
-        # and if today were December 5th, 2020, 'relative' would be "1
-        # month".
-        date_format = _(u"{relative} ago - {absolute}") if date_has_passed else _(u"in {relative} - {absolute}")
-        return date_format.format(
-            relative=relative_date,
-            absolute=self.date.astimezone(self.time_zone).strftime(self.date_format.encode('utf-8')).decode('utf-8'),
-        )
+        if date_has_passed:
+            return relative_date + " ago - {date}"
+        else:
+            return "in " + relative_date + " - {date}"
```

Comment:
I see you attempt to translate this in the mako template.  However, this won't work as you are intending.  The words "ago" and "in" are located here.  But the translation happens elsewhere.  We can discuss more tomorrow.